sdk/core/azure-core/tests/test_messaging_cloud_event.py

```diff
@@ -0,0 +1,128 @@
+import logging
+import sys
+import os
+import pytest
+import json
+from datetime import datetime
+
+from azure.core.messaging import CloudEvent
+
+# Cloud Event tests
+def test_cloud_event_constructor():
+    event = CloudEvent(
+        source='Azure.Core.Sample',
+        type='SampleType',
+        data='cloudevent'
+        )
+    
+    assert event.specversion == '1.0'
+    assert event.time.endswith('+00:00')
+    assert event.id is not None
+    assert event.source == 'Azure.Core.Sample'
+    assert event.data == 'cloudevent'
+
+def test_cloud_storage_dict():
+    cloud_storage_dict = {
+        "id":"a0517898-9fa4-4e70-b4a3-afda1dd68672",
+        "source":"/subscriptions/{subscription-id}/resourceGroups/{resource-group}/providers/Microsoft.Storage/storageAccounts/{storage-account}",
+        "data":{
+            "api":"PutBlockList",
+            "client_request_id":"6d79dbfb-0e37-4fc4-981f-442c9ca65760",
+            "request_id":"831e1650-001e-001b-66ab-eeb76e000000",
+            "e_tag":"0x8D4BCC2E4835CD0",
+            "content_type":"application/octet-stream",
+            "content_length":524288,
+            "blob_type":"BlockBlob",
+            "url":"https://oc2d2817345i60006.blob.core.windows.net/oc2d2817345i200097container/oc2d2817345i20002296blob",
+            "sequencer":"00000000000004420000000000028963",
+            "storage_diagnostics":{"batchId":"b68529f3-68cd-4744-baa4-3c0498ec19f0"}
+        },
+        "type":"Microsoft.Storage.BlobCreated",
+        "time":"2020-08-07T01:11:49.765846Z",
+        "specversion":"1.0"
+    }
+
+    event = CloudEvent.from_dict(cloud_storage_dict)
+    assert event.data == {
+        "api":"PutBlockList",
+        "client_request_id":"6d79dbfb-0e37-4fc4-981f-442c9ca65760",
+        "request_id":"831e1650-001e-001b-66ab-eeb76e000000",
+        "e_tag":"0x8D4BCC2E4835CD0",
+        "content_type":"application/octet-stream",
+        "content_length":524288,
+        "blob_type":"BlockBlob",
+        "url":"https://oc2d2817345i60006.blob.core.windows.net/oc2d2817345i200097container/oc2d2817345i20002296blob",
+        "sequencer":"00000000000004420000000000028963",
+        "storage_diagnostics":{"batchId":"b68529f3-68cd-4744-baa4-3c0498ec19f0"}
+    }
+    assert event.specversion == "1.0"
```

Comment:
Would expect to see a time test on datetime

Reply:
added